### PR TITLE
hygiene :rebind loop variable and wrap the actual test logic inside in suite.Run()

### PR DESCRIPTION
**Pull request by trinitys7**
## Description
- rebind loop variable in tests
- wrap the actual test logic inside in suite.Run()

closes: #4419 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/capability/simulation/decoder_test.go [hunk 1]
```diff
@@ -48,14 +48,14 @@ func TestDecodeStore(t *testing.T) {
...
-		t.Run(tt.name, func(t *testing.T) {
+	for i, tc := range tests {
+		i, tc := i, tc
+		t.Run(tc.name, func(t *testing.T) {
 			switch i {
 			case len(tests) - 1:
 				require.Panics(t, func() { dec(kvPairs.Pairs[i], kvPairs.Pairs[i]) }, tt.name)
 			default:
-				require.Equal(t, tt.expectedLog, dec(kvPairs.Pairs[i], kvPairs.Pairs[i]), tt.name)
+				require.Equal(t, tc.expectedLog, dec(kvPairs.Pairs[i], kvPairs.Pairs[i]), tt.name)
```
Comment (damiannolan) [inline modules/capability/simulation/decoder_test.go:58]: `tt.name` still referenced in `require` assertion, should be `tc` now

**Review by damiannolan — COMMENTED** [review]
Thanks for opening this PR @trinitys7, just left one initial comment as CI job for capability tests are failing.

File: modules/capability/simulation/decoder_test.go (hunk 1 above)
Comment (trinitys7) [reply modules/capability/simulation/decoder_test.go]: fixed !

**Review by crodriguezvega — COMMENTED** [review]
Thank you, @trinitys7. One comment for now.

**Review by damiannolan — APPROVED** [review]
Thank you @trinitys7!

**Review by DimitrisJim — APPROVED** [review]
lgtm considering comments left in other reviews

File: modules/apps/transfer/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -73,20 +73,23 @@ func (suite *KeeperTestSuite) TestQueryDenomTrace() {
 
 	for _, tc := range testCases {
 		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
-			suite.SetupTest() // reset
+			tc := tc
+			suite.Run(tc.msg, func() {
```
[the remark this answers is not in the document]
Comment (trinitys7) [reply modules/apps/transfer/keeper/grpc_query_test.go:77]: yes, it is duplicated. I'm fixing it now !

**Comment by crodriguezvega** [comment]
@trinitys7 if you can rebase with main, then we will merge the PR. Thank you!

**Pull request merged**